cmd/podman/common/specgen.go

```diff
@@ -252,18 +252,21 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 	}
 	s.Expose = expose
 
-	for k, v := range map[string]*specgen.Namespace{
-		c.IPC:       &s.IpcNS,
-		c.PID:       &s.PidNS,
-		c.UTS:       &s.UtsNS,
-		c.CGroupsNS: &s.CgroupNS,
-	} {
-		if k != "" {
-			*v, err = specgen.ParseNamespace(k)
```

Comment:
This looks like a classic loop-variable overwrite/leak. Good catch!

Reply:
I think the issue was that there are collisions with the keys (both c.IPC and c.PID have the value "host")

both